cmd/repo-init/frontend/src/app/BuildConfig/BaseImages.tsx

```diff
@@ -0,0 +1,176 @@
+} from "@patternfly/react-core";
+import {AuthContext, ConfigContext, Image, WizardContext} from "@app/types";
+import {ErrorMessage} from "@app/Common/Messaging";
+import {validateConfig} from "@app/utils/utils";
+import _ from "lodash";
+
+const BaseImages: React.FunctionComponent = () => {
+  const columns = ['Name', 'Namespace', 'Tag', '']
+  const authContext = useContext(AuthContext);
+  const configContext = useContext(ConfigContext);
+  const context = useContext(WizardContext);
+
+  const [curImage, setCurImage] = useState({} as Image)
+  const [errorMessage, setErrorMessage] = useState([] as string[])
+
+  function handleChange(val, evt) {
+    const updated = {...curImage}
+    _.set(updated, evt.target.name, val);
+    setCurImage(updated);
+  }
+
+  function saveImage() {
+    if (curImage.name && curImage.namespace && curImage.tag) {
+      const config = configContext.config;
+      const buildConfig = config.buildSettings;
+      if (buildConfig) {
+        if (!buildConfig.baseImages) {
+          buildConfig.baseImages = [];
+        }
+        if (buildConfig.baseImages.find(t => (t.name.toLowerCase() === curImage.name.toLowerCase()) && (t.namespace.toLowerCase() === curImage.namespace.toLowerCase()) && (t.tag.toLowerCase() === curImage.tag.toLowerCase())) === undefined) {
+          const updatedBaseImages = config.buildSettings.baseImages.concat(curImage);
+          const updatedConfig = {...config, buildSettings: {...config.buildSettings, baseImages: updatedBaseImages}};
+          validate(updatedConfig);
+        } else {
+          setErrorMessage(["That base image already exists"])
+        }
+      }
+    } else {
+      setErrorMessage(["Please provide a name, namespace, and tag for the image."])
+    }
+  }
+
+  function validate(validationConfig) {
+    validateConfig('BASE_IMAGES', validationConfig, authContext.userData, {})
+      .then((validationState) => {
+        if (validationState.valid) {
+          setErrorMessage([]);
+          configContext.setConfig(validationConfig);
+          setCurImage({} as Image);
+          context.setStep({
+            ...context.step,
+            errorMessages: [],
+            stepIsComplete: true
+          });
+        } else {
+          setErrorMessage(validationState.errors != undefined ? validationState.errors.map(error => error.message) : [""]);
+          context.setStep({
+            ...context.step,
+            stepIsComplete: false
+          });
+        }
+      })
+  }
+
+  function removeImage(index) {
+    const config = configContext.config;
+    const buildConfig = config.buildSettings;
+    if (buildConfig && buildConfig.baseImages) {
+      const updatedBaseImages = [...buildConfig.baseImages];
+      updatedBaseImages.splice(index, 1);
+      const updatedConfig = {...config, buildSettings: {...config.buildSettings, baseImages: updatedBaseImages}};
+      configContext.setConfig(updatedConfig);
+    }
+  }
+
+  const SummaryTable = () => {
```

Comment:
Would be good to ask someone on e.g. Sam Padgett's team or generally in the UI space toreview

Reply:
Yeah, I'm definitely going to get them to review this.